src/ILCompiler.RyuJit/src/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -943,5 +898,34 @@ private CorInfoHelpFunc getCastingHelper(ref CORINFO_RESOLVED_TOKEN pResolvedTok
 
             return helper;
         }
+
+        private CorInfoHelpFunc getNewHelper(ref CORINFO_RESOLVED_TOKEN pResolvedToken, CORINFO_METHOD_STRUCT_* callerHandle)
+        {
+            TypeDesc type = HandleToObject(pResolvedToken.hClass);
+
+            Debug.Assert(!type.IsString && !type.IsArray && !type.IsCanonicalDefinitionType(CanonicalFormKind.Any));
+
+            // TODO: disambiguate to the specialized helpers we have (see GetNewObjectHelperForType)
```

Comment:
Do we have any other options than to introduce a new CorInfoHelpFunc, update JitInterfaceGuid and integrate a new RyuJIT here?

Reply:
That's how we have been modelling the helpers so far, and it has been less than ideal in various ways. Introducing new  CorInfoHelpFuncs as you have suggested is probably the best option right now. I suspect that the refactoring required for this would be too disruptive.

The discussion in https://github.com/dotnet/coreclr/issues/21530 is related to this.